src/glue/auth_checker.cpp

```diff
@@ -114,6 +132,26 @@ bool AuthChecker::IsUserAuthorized(const memgraph::auth::User &user,
   });
 }
 
+bool AuthChecker::IsUserAdmin(const memgraph::auth::User &user) {
+  const auto user_permissions = user.GetPermissions();
+  const std::vector<memgraph::query::AuthQuery::Privilege> privileges{
+      memgraph::query::AuthQuery::Privilege::CREATE,      memgraph::query::AuthQuery::Privilege::DELETE,
+      memgraph::query::AuthQuery::Privilege::MATCH,       memgraph::query::AuthQuery::Privilege::MERGE,
+      memgraph::query::AuthQuery::Privilege::SET,         memgraph::query::AuthQuery::Privilege::REMOVE,
+      memgraph::query::AuthQuery::Privilege::INDEX,       memgraph::query::AuthQuery::Privilege::STATS,
+      memgraph::query::AuthQuery::Privilege::AUTH,        memgraph::query::AuthQuery::Privilege::CONSTRAINT,
+      memgraph::query::AuthQuery::Privilege::DUMP,        memgraph::query::AuthQuery::Privilege::REPLICATION,
+      memgraph::query::AuthQuery::Privilege::DURABILITY,  memgraph::query::AuthQuery::Privilege::READ_FILE,
+      memgraph::query::AuthQuery::Privilege::FREE_MEMORY, memgraph::query::AuthQuery::Privilege::TRIGGER,
+      memgraph::query::AuthQuery::Privilege::CONFIG,      memgraph::query::AuthQuery::Privilege::STREAM,
+      memgraph::query::AuthQuery::Privilege::MODULE_READ, memgraph::query::AuthQuery::Privilege::MODULE_WRITE,
+      memgraph::query::AuthQuery::Privilege::WEBSOCKET};
```

Comment:
I have two thoughts about this. First, this seems not maintainable because if we change enum class `Privilege` we need to check that this vector is also updated. Second thought is that we can make this static array that it is accessed from other parts of code.

I think we can do the following to make it less bug prone.

Create following array:

static constexpr Privilege privilege_array[] = {
  Privilege::CREATE,
  Privilege::DELETE,
  // ... and so on
};

static_assert(std::size(privilege_array) == static_cast<std::size_t>(Privilege::WEBSOCKET), "Missing Privilege values in privilege_array");

And afterwards, you can initialize this

Reply:
with following :
const std::vector<Privilege> privileges{
  std::begin(privilege_array),
  std::end(privilege_array)
};

Reply:
Yes, assert should help under assumption we don't assign specific values to enum.

Reply:
AhI just realized we already have kPrivilegesAll in ast.hpp